### PR TITLE
test: add test for parseFromUrl function

### PR DESCRIPTION
**Pull request by aayushmau5**
**Description**

- Add test for `parseFromUrl` function present in `parser.js` file.
- Wrap the exception thrown by `parseFromUrl` function in `parserError` object.
- Modified the `package.json` scripts to combine reports from multiple tests.
- Update README to add the Error Type.

### CODE REVIEW

File: lib/parser.js [hunk 1]
```diff
@@ -133,7 +133,13 @@ function parseFromUrl(url, fetchOptions, options) {
       .then(res => res.text())
       .then(doc => parse(doc, options))
       .then(result => resolve(result))
-      .catch(reject);
+      .catch(e => {
+        if (e instanceof ParserError) return reject(e);
+        return reject(new ParserError({
+          type: 'unexpected-error',
```
Comment (derberg) [inline lib/parser.js:139]: shouldn't we maybe introduce a custom type here, like maybe `url-fetching-error` or `fetching-error`?
if you agree, don't forget about updating the readme

File: package.json [hunk 2]
```diff
@@ -17,7 +17,9 @@
     "get:version": "echo $npm_package_version",
     "get:name": "echo $npm_package_name",
     "lint": "eslint --max-warnings 0 --config \".eslintrc\" \".\"",
-    "test:lib": "nyc --reporter=html --reporter=text mocha --exclude \"test/browser_test.js\" --recursive",
+    "test:lib": "nyc --silent --no-clean mocha --exclude \"test/browser_test.js\" --exclude \"test/parseFromUrl_test.js\" --recursive",
+    "test:parseFromUrl":"nyc --silent --no-clean start-server-and-test \"http-server test/sample_browser --cors -s\" 8080 \"mocha test/parseFromUrl_test.js\"",
```
Comment (derberg) [inline package.json:21]: I like you are reusing the sample browser from another test

**Review by derberg — COMMENTED** [review]
Well done mate. Statements coverate for parser.js jumped from 91 to 99, just one error is not yet tested 🥳 

Just one question from my side, have a look.

File: package.json (hunk 2 above)
Comment (aayushmau5) [reply package.json:21]: Yup, all thanks to your for giving me pointers.

File: lib/parser.js (hunk 1 above)
Comment (aayushmau5) [reply lib/parser.js:139]: Agree. `fetch-url-error` sounds good. Also, thanks for reminding to update the readme.

Comment (derberg) [reply lib/parser.js:139]: awesome, so go ahead with the change, in the test also validate not only message but type too please

**Comment by aayushmau5** [comment]
@derberg Please check :)

**Review by derberg — APPROVED** [review]

**Comment by derberg** [comment]
@all-contributors please add @aayushmau5 for test

**Pull request merged**